test/src/test/java/org/corfudb/runtime/collections/CorfuQueueTest.java

```diff
@@ -1,38 +1,58 @@
 package org.corfudb.runtime.collections;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+
 import com.google.common.primitives.UnsignedBytes;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.Assertions;
+import org.corfudb.runtime.CorfuRuntime;
 import org.corfudb.runtime.collections.CorfuQueue.CorfuQueueRecord;
 import org.corfudb.runtime.view.AbstractViewTest;
 import org.corfudb.util.serializer.Serializers;
 import org.junit.Test;
 
-import java.util.Comparator;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-
 /**
  * Created by Sundar Sridharan on May 22, 2019
  */
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /**
  * Simple test of basic operations to check that insert order is preserved in the queue.
  * Created by hisundar on 05/27/2019
  */
+@Slf4j
 public class CorfuQueueTest extends AbstractViewTest {
 
+    @Test
+    public void failNonTxnEnqueue() {
+        CorfuQueue<String>
+                corfuQueue = new CorfuQueue<>(getDefaultRuntime(), "test");
+        Assertions.assertThatThrownBy(() -> corfuQueue.enqueue("c"))
+                .isInstanceOf(IllegalStateException.class)
```

Comment:
Just a question: Are we ok with IllegalStateException? Maybe we have a more specific exception?

Reply:
I think this should suffice. This exception signals an incorrect usage of the API. There is nothing to handle per-se
```java

 * Signals that a method has been invoked at an illegal or
 * inappropriate time.  In other words, the Java environment or
 * Java application is not in an appropriate state for the requested
 * operation.

```